emperor/support_files/js/color-view-controller.js

```diff
@@ -128,48 +173,79 @@ define([
    * category in a given metadata column.
    * @param {map} name of the color map to use, see
    * ColorViewController.Colormaps
+   * @param {discrete} Whether to treat colormap requested as a discrete set
+   * of colors or use interpolation to create gradient of colors
+   * @param {scaled} Whether to use a scaled colormap or equidistant colors for
+   * each value. Defaiult false.
    *
    *
    * This function will generate a list of coloring values depending on the
    * coloring scheme that the system is currently using (discrete or
    * continuous).
    */
-  ColorViewController.getColorList = function(values, map) {
-    var colors = {}, numColors = values.length-1, counter=0, discrete = false;
-    var interpolator;
-
-    if (ColorViewController.Colormaps.indexOf(map) === -1) {
-      throw new Error("Could not find "+map+" in the available colormaps");
-    }
+  ColorViewController.getColorList = function(values, map, discrete, scaled) {
+    var colors = {}, numColors = values.length - 1, counter = 0;
+    var interpolator, min, max;
+    var scaled = scaled || false;
 
-    if (map === 'discrete-coloring' || map === 'discrete-coloring-qiime'){
-      discrete = true;
+    if (_.findWhere(ColorViewController.Colormaps, {id: map}) === undefined){
+      throw new Error("Could not find "+map+" as a colormap.");
     }
 
     // 1 color and continuous coloring should return the first element of the
     // map
     if (numColors === 0 && discrete === false){
       colors[values[0]] = chroma.brewer[map][0];
-      return colors;
+      return [colors, gradientSVG];
     }
 
-    if (discrete === false){
+    if (discrete === false) {
       map = chroma.brewer[map];
       interpolator = chroma.bezier([map[0], map[3], map[4], map[5], map[8]]);
+      if (scaled === true) {
+        //assuming we have numeric since scaled called
+        var numericValues = _.map(values, Number);
+        min = _.min(numericValues);
+        max = _.max(numericValues);
+        interpolator = chroma.scale(map).domain([min, max]);
+      }
     }
 
-    for(var index in values){
-      if(discrete){
+    for (var index in values) {
+      if (discrete) {
         // get the next available color
         colors[values[index]] = ColorViewController.getDiscreteColor(index, map);
       }
-      else{
-        colors[values[index]] =  interpolator(counter/numColors).hex();
+      else if (scaled === true) {
+        colors[values[index]] = interpolator(Number(values[index])).hex();
+      } else {
+        colors[values[index]] = interpolator(counter / numColors).hex();
         counter = counter + 1;
       }
     }
 
-    return colors;
+    //if scaled, generate the scale and add to div.
+    if (scaled) {
+      var min = _.min(numericValues);
+      var max = _.max(numericValues);
+      var mid = (min + max) / 2;
+      step = (max - min) / 100;
+      var stopColors = [];
+      for (var s = min; s <= max; s += step) {
```

Comment:
I believe you can replace some of this with underscore's range, see here:
http://underscorejs.org/#range